binderhub/builder.py

```diff
@@ -55,6 +56,30 @@
 LAUNCHES_INPROGRESS = Gauge('binderhub_inprogress_launches', 'Launches currently in progress')
 
 
+def _get_image_basename_and_tag(full_name):
+    """Get a supposed image name and tag without the registry part
+    :param full_name: full image specification, e.g. "gitlab.com/user/project:tag"
+    :return: tuple of image name and tag, e.g. ("user/project", "tag")
+    """
+    # the tag is either after the last (and only) colon, or not given at all,
+    # in which case "latest" is implied
+    tag_splits = full_name.rsplit(':', 1)
+    if len(tag_splits) == 2:
+        image_name = tag_splits[0]
+        tag = tag_splits[1]
+    else:
+        image_name = full_name
+        tag = 'latest'
```

Comment:
In the current implementation, I think the tag is left unspecified if it isn't present. Let's preserve that?

Reply:
I wouldn't mind. If i see correctly, the tag is passed to `registry.get_image_manifest()` where the following happens:
`"{}/v2/{}/manifests/{}".format(self.url, image, tag)`
So if tag is an empty string, could this be a problem for the manifests API? Idk personally, as i am pretty much clueless about this.

Reply:
@rappertomate I think tag is just always required to be specified when pushing, and is never left empty. I wouldn't mind raising an exception if tag is empty or None, but we can tackle that in another PR. I mostly just wanted to make sure the behavior doesn't change.

I think once that's done I'll happily merge this.